transport/network/src/ping.rs

```diff
@@ -77,61 +78,111 @@ pub struct PingConfig {
     pub timeout: std::time::Duration, // `Duration` -> should be in millis,
 }
 
-/// Trait for the ping operation itself.
-#[async_trait]
-pub trait Pinging {
-    async fn ping(&mut self, peers: Vec<PeerId>);
+pub type PingQueryResult = std::result::Result<(std::time::Duration, String), ()>;
+
+#[derive(Debug)]
+pub struct PingQueryReplier {
```

Comment:
All `pub` stuff should be documented, types and their methods too. But we can also leave this for docs day.